### ISSUE TITLE
question on Intra_Patch_Attention Class

### ISSUE THREAD

**Issue by efredGN** [decisionintelligence/pathformer#12]
in forward method, 
query = torch.cat(torch.split(query, self.head_size, dim=-1), dim=0)
key = torch.cat(torch.split(key, self.head_size, dim=-1), dim=0)
value = torch.cat(torch.split(value, self.head_size, dim=-1), dim=0)
query = query.permute((0, 2, 1, 3))

Does it mean we have 4 dimension in query after torch.cat implements? I am confused with it. Usually, query should have 3 dimension after torch.cat method. If possible, please explain it, thanks
